app/controllers/organizations/tasks_controller.rb

```diff
@@ -25,6 +25,7 @@ def edit
 
   def update
     if @task.update(task_params)
+      Organizations::TaskService.new(@task).create_next
```

Comment:
Ok cool I see what you are doing here. I wonder, for sake of readability on the controller, if we want to check if a new task needs to be created in the Controller, then instantiate the service? Currently, this will instantiate the service regardless, e.g., when a typo is fixed in the name or description, and this is not a necessary step. So, we could add a conditional that calls the service object if `@task.recurring && @task.completed_previously_changed?(from: false, to: true)` on update. Does that make sense?

Reply:
I was so on the fence about whether to err on the side of keeping the controller code minimal or not, but I think you're right that it's better to keep the logic of it up front.

Reply:
Yes I think a controller is going to be read more often, so it's better if it tells the story, rather than have to look in the Service object to figure out what is going on.